nixos/modules/services/desktops/accountsservice.nix

```diff
@@ -36,11 +36,14 @@ with lib;
 
     systemd.packages = [ pkgs.accountsservice ];
 
-    systemd.services.accounts-daemon= {
+    systemd.services.accounts-daemon = {
 
       wantedBy = [ "graphical.target" ];
 
-    } // (mkIf (!config.users.mutableUsers) {
+      # Accounts daemon looks for dbus interfaces in $XDG_DATA_DIRS/accountsservice
+      environment.XDG_DATA_DIRS = "/run/current-system/sw/share";
```

Comment:
I think something like `${config.system.path}/share` should be better.

Reply:
Done. Thank you for your review!

Reply:
Hm, shouldn't systemd system services run in an environment that has `XDG_DATA_DIRS` set appropriately for the current system? I'm surprised more things don't break if that not true.

Reply:
The variables are often set by login shell, so the services cannot see them. It is probably good thing, as the unclean environment could produce unpredictable behaviour.